functionary/ui/views/schedules/utils.py

```diff
@@ -51,6 +51,19 @@ def get_or_create_crontab_schedule(
     return crontab_schedule
 
 
+def get_crontab_schedule(schedule_form_data: dict) -> CrontabSchedule:
```

Comment:
This function belongs here, but the rest of these were more appropriate living in `core/utils/scheduling.py`

The distinction is that this function deals with UI specific stuff: it's parsing the form data and using it call the general `get_or_create_crontab_schedule`.

The rest of the functions in this file have applicability outside of the UI and could conceivably called in other contexts (as part of API request handline for instance).

Basically, leave the old utils file as it was and create this as a new file here with your new function.

Reply:
Fixed